File: Pila.java
package pila;
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        arreglo = new long[n];
        top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("Pila llena");
        } else {
            arreglo[++top] = e;
        }
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("Pila vacia");
            return -1;
        } else {
            return arreglo[top--];
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Pila vacia");
            return -1;
        } else {
            return arreglo[top];
        }
    }

    public boolean isEmpty() {
        return top == -1; 
    }

    public boolean isFull() {
        return top == n - 1; 
    }

    public static void main(String[] args) {
        Pila pila = new Pila(3);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        System.out.println(pila.peek()); 
        System.out.println(pila.pop());  
        System.out.println(pila.peek()); 
        System.out.println(pila.pop());
        System.out.println(pila.pop());
    }
}

File: Pila.py
class Pila:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.top = -1

    def push(self, e):
        if self.is_full():
            print("Pila llena")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if self.is_empty():
            print("Pila vacia")
            return -1
        else:
            elemento = self.arreglo[self.top]
            self.top -= 1
            return elemento

    def peek(self):
        if self.is_empty():
            print("Pila vacia")
            return -1
        else:
            return self.arreglo[self.top]

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return self.top == self.n - 1




if __name__ == "__main__":
    pila = Pila(3)
    pila.push(10)
    pila.push(20)
    pila.push(30)
    print(pila.peek())  
    print(pila.pop())   
    print(pila.peek())  
    print(pila.pop())   
    print(pila.pop())
    print(pila.pop())
